flag_timestamp.go

```diff
@@ -0,0 +1,130 @@
+package cli
+
+import (
+	"flag"
+	"fmt"
+	"time"
+)
+
+
+// timestamp wrap to satisfy golang's flag interface.
+type timestampWrap struct {
+	timestamp *time.Time
+	hasBeenSet bool
+	layout 	   string
+}
+
+// Set the timestamp value directly
+func (t *timestampWrap) SetTimestamp(value time.Time) {
+	if !t.hasBeenSet {
+		t.timestamp = &value
+		t.hasBeenSet = true
+	}
+}
+// Set the timestamp string layout for future parsing
+func (t *timestampWrap) SetLayout(layout string) {
+	t.layout = layout
+}
+
+// Parses the string value to timestamp
+func (t *timestampWrap) Set(value string) error {
+	timestamp, err := time.Parse(t.layout, value)
+	if err != nil {
+		return err
+	}
+
+	t.timestamp = &timestamp
+	return nil
+}
+
+// String returns a readable representation of this value (for usage defaults)
+func (t *timestampWrap) String() string {
+	return fmt.Sprintf("%#v", t.timestamp)
+}
+
+// Value returns the timestamp value stored in the flag
+func (t *timestampWrap) Value() *time.Time {
+	return t.timestamp
+}
+
+// Get returns the flag structure
+func (t *timestampWrap) Get() interface{} {
+	return *t
+}
+
+// TimestampFlag is a flag with type protobuf.timestamp
+type TimestampFlag struct {
+	Name        string
+	Aliases     []string
+	Usage       string
+	EnvVars     []string
+	FilePath    string
+	Required    bool
+	Hidden      bool
+	Layout 	    string
+	Value       timestampWrap
```

Comment:
```suggestion
	Value       *timestampWrap
```

This should be a pointer to the wrapped struct